### ISSUE TITLE
unable to see cursor with base16_terminal theme

### ISSUE THREAD

**Issue by LollipopFt** [helix-editor/helix#1564]
### Reproduction steps
open a text file, notice there is no cursor which can be seen.

https://user-images.githubusercontent.com/62802897/150666218-5e0cec9c-af4d-4b3b-8564-e24b23cad23e.mp4

 

### Environment

- Platform: 
- Terminal emulator: 
- Helix version: 

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

**Comment by NNBnh** [comment]
Fixed here: https://github.com/helix-editor/helix/pull/1525/commits/183914180eddf2241f2d806d8b9ac8481056370c

**Comment by tasmo** [comment]
> Fixed here: [1839141](https://github.com/helix-editor/helix/commit/183914180eddf2241f2d806d8b9ac8481056370c)

Unfortunately this just fixes the issue on dark terminals and not on light backgrounds.

**Comment by NNBnh** [comment]
> Unfortunately this just fixes the issue on dark terminals and not on light backgrounds.

- **Previous problem:** from #1078:
  > Some options with "white" foreground have a `#FIXME` comment at the end, that because the intent foreground color is "default" foreground color (which currently does not exist) not "white".
  - Because the "default" color doesn't exist the only workaround is to not set it entirely so it will fallback to using the terminal foreground color.
  - Unfortunately, this cause the cursor color to fall back to the color under it, make it change with the syntax highlight, some time cause it to become invisible.
- **Current problem:** after [1839141](https://github.com/helix-editor/helix/commit/183914180eddf2241f2d806d8b9ac8481056370c):
  - Although this is not correct to the theme design, I have to hotfix it by set the cursor back to "white".
  - And of course, this lead to light background terminal not able to see the cursor...

We could set the cursor to a specific color (like blue) but I strongly against this idea because it break the neutral design of the theme.
